skfda/representation/basis.py

```diff
@@ -1482,6 +1482,17 @@ def penalty(self, derivative_degree=None, coefficients=None):
             # implement using inner product
             return self._numerical_penalty(coefficients)
 
+    def gram_matrix(self):
```

Comment:
This is only true if the period and the domain range are the same. Otherwise the basis is not orthonormal

Reply:
I see. So check if period and domain_range are the same. If so return identity and otherwise return super.gram_matrix()?

Reply:
That sounds good. Remember that the period must be equal to the difference of the limits in the domain range.